app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/MainActivity.kt

```diff
@@ -0,0 +1,493 @@
+        }
+    }
+
+    fun fetchFahrplan() {
+        if (MyApp.task_running == TASKS.NONE) {
+            MyApp.task_running = TASKS.FETCH
+            showFetchingStatus()
+            val url = appRepository.readScheduleUrl()
+            appRepository.loadSchedule(
+                url = url,
+                onFetchingDone = ::onGotResponse,
+                onParsingDone = ::onParseDone,
+                onLoadingShiftsDone = ::onLoadShiftsDone
+            )
+        } else {
+            Log.d(LOG_TAG, "Fetching schedule already in progress.")
+        }
+    }
+
+    override fun onDestroy() {
+        appRepository.cancelLoading()
+        hideProgressDialog()
+        super.onDestroy()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        isScreenLocked = keyguardManager.isKeyguardLocked
+        val sidePaneView = findViewById<FragmentContainerView>(R.id.detail)
+        if (sidePaneView != null && isFavoritesInSidePane) {
+            sidePaneView.isVisible = !isScreenLocked
+        }
+        if (!appRepository.readScheduleChangesSeen()) {
+            showChangesDialog()
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        super.onCreateOptionsMenu(menu)
+        menuInflater.inflate(R.menu.mainmenu, menu)
+        return true
+    }
+
+    private fun showChangesDialog() {
+        val fragment = findFragment(ChangesDialog.FRAGMENT_TAG)
+        if (fragment == null) {
+            val sessions = appRepository.loadChangedSessions()
+            val meta = appRepository.readMeta()
+            val statistic = ChangeStatistic.of(sessions)
+            ChangesDialog
+                .newInstance(meta.version, statistic)
+                .show(supportFragmentManager, ChangesDialog.FRAGMENT_TAG)
+        }
+    }
+
+    private fun showAboutDialog() {
+        val meta = appRepository.readMeta()
+        val transaction = supportFragmentManager.beginTransaction()
+        transaction.addToBackStack(null)
+        AboutDialog
+            .newInstance(meta.version, meta.subtitle, meta.title)
+            .show(transaction, AboutDialog.FRAGMENT_TAG)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.menu_item_about -> {
+                showAboutDialog()
+                return true
+            }
+            R.id.menu_item_alarms -> {
+                AlarmList.startForResult(this)
+                return true
+            }
+            R.id.menu_item_settings -> {
+                SettingsActivity.startForResult(this)
+                return true
+            }
+            R.id.menu_item_schedule_changes -> {
+                openSessionChanges()
+                return true
+            }
+            R.id.menu_item_favorites -> {
+                openFavorites()
+                return true
+            }
+        }
+        return super.onOptionsItemSelected(item)
```

Comment:
Maybe consider:

```
val actionMap = mapOf(
    R.id.menu_item_about to { showAboutDialog() },
    R.id.menu_item_alarms to { AlarmList.startForResult(this) },
    R.id.menu_item_settings to { SettingsActivity.startForResult(this) },
    R.id.menu_item_schedule_changes to { openSessionChanges() },
    R.id.menu_item_favorites to { openFavorites() }
)
val itemId = item.itemId

return if (actionMap.containsKey(itemId)) {
    actionMap[itemId]!!()
    true
} else {
    super.onOptionsItemSelected(item)
}
```

Reply:
Did you try this? Is `this` already valid when the `actionMap` property is initialized?

Reply:
I try it only in the scope of the function, which means  I did not add `actionMap` (which might need a better name) as property.  However it should be possible too. On a 2nd thought - do you know if `onOptionsItemSelected ` has a case where it goes `false`? If not you can go even further with `getOrElse` I suppose.

Reply:
:heavy_check_mark: Changed in fd9d3aae9a775f6388e5aade8779c1ce15289c06.